File: frontend/src/types/patient.ts
export type SexEnum = 'male' | 'female';
export type SmokingStatusEnum = 'never' | 'former' | 'current' | 'unknown';

export interface Medication {
  name: string;
  dosage: string;
  frequency: string;
  start_date?: string;
  instructions?: string;
}

export interface SurgicalProcedure {
  procedure: string;
  date: string;
  notes?: string;
}

export interface SmokingDetails {
  pack_years?: number;
  quit_date?: string;
}

export interface PatientBase {
  name: string;
  national_id: string;
  date_of_birth: string;
  phone_primary: string;
  phone_secondary?: string;
  email?: string;
  address?: string;
  sex: SexEnum;
  chronic_diseases: string[];
  allergies: string[];
  current_medications: Medication[];
  surgical_history: SurgicalProcedure[];
  smoking_status: SmokingStatusEnum;
  smoking_details?: SmokingDetails;
}

export interface Patient extends PatientBase {
  patient_id: string;
  age: number;
  registration_date: string;
  last_updated: string;
  last_updated_by: string;
  total_sessions: number;
  last_session_id?: string;
  last_session_date?: string;
}

export interface PatientSearchResult {
  patient_id: string;
  name: string;
  age: number;
  phone_primary: string;
  national_id: string;
  last_session_date?: string;
  total_sessions: number;
}

export interface PatientCreateRequest extends PatientBase {}

export interface PatientUpdateRequest extends Partial<PatientBase> {}

File: backend/app/models/patient.py
from pydantic import BaseModel, EmailStr, Field
from typing import Optional, List
from datetime import date, datetime
from enum import Enum


class SexEnum(str, Enum):
    male = "male"
    female = "female"


class SmokingStatusEnum(str, Enum):
    never = "never"
    former = "former"
    current = "current"
    unknown = "unknown"


class Medication(BaseModel):
    name: str
    dosage: str
    frequency: str
    start_date: Optional[date] = None
    instructions: Optional[str] = None


class SurgicalProcedure(BaseModel):
    procedure: str
    date: date
    notes: Optional[str] = None


class SmokingDetails(BaseModel):
    pack_years: Optional[int] = None
    quit_date: Optional[date] = None


class PatientBase(BaseModel):
    # I. Personal Identification (Mandatory)
    name: str = Field(..., min_length=2, max_length=200)
    national_id: str = Field(..., min_length=10, max_length=20)
    date_of_birth: date
    phone_primary: str = Field(..., min_length=10, max_length=20)
    
    # II. Optional Contact Information
    phone_secondary: Optional[str] = Field(None, min_length=10, max_length=20)
    email: Optional[EmailStr] = None
    address: Optional[str] = None
    
    # III. Medical Background
    sex: SexEnum
    chronic_diseases: List[str] = []
    allergies: List[str] = []
    current_medications: List[Medication] = []
    surgical_history: List[SurgicalProcedure] = []
    smoking_status: SmokingStatusEnum = SmokingStatusEnum.unknown
    smoking_details: Optional[SmokingDetails] = None


class PatientCreate(PatientBase):
    pass


class PatientUpdate(BaseModel):
    # All fields optional for updates
    name: Optional[str] = Field(None, min_length=2, max_length=200)
    date_of_birth: Optional[date] = None
    phone_primary: Optional[str] = Field(None, min_length=10, max_length=20)
    phone_secondary: Optional[str] = Field(None, min_length=10, max_length=20)
    email: Optional[EmailStr] = None
    address: Optional[str] = None
    sex: Optional[SexEnum] = None
    chronic_diseases: Optional[List[str]] = None
    allergies: Optional[List[str]] = None
    current_medications: Optional[List[Medication]] = None
    surgical_history: Optional[List[SurgicalProcedure]] = None
    smoking_status: Optional[SmokingStatusEnum] = None
    smoking_details: Optional[SmokingDetails] = None


class Patient(PatientBase):
    patient_id: str
    age: int
    registration_date: datetime
    last_updated: datetime
    last_updated_by: str
    total_sessions: int = 0
    last_session_id: Optional[str] = None
    last_session_date: Optional[datetime] = None
    
    class Config:
        from_attributes = True


class PatientSearchResult(BaseModel):
    patient_id: str
    name: str
    age: int
    phone_primary: str
    national_id: str
    last_session_date: Optional[datetime] = None
    total_sessions: int

